### PR TITLE
Add cli artifacts Dockerfile

### PR DESCRIPTION
**Pull request by navidshaikh**
- Add Makefile target build-cross-package that builds cross platform
   binaries and package them as tar gz for Linux and macOS and as zip for Windows.
   Each package contains a LICENSE with binary.

 - Add Dockerfile.cliartifacts which includes the tar gzs and zip files in image.

 - Later this image can be used as deployment and have the tar gzs and zip served from image

### CODE REVIEW

File: openshift/ci-operator/knative-images/client/Dockerfile.cliartifacts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+
+RUN yum -y install zip && yum clean all
+RUN bash package_cliartifacts.sh
+RUN mkdir -p /usr/share/kn/{linux_amd64,macos,windows} &&
+    mv kn-linux-amd64.tar.gz /usr/share/kn/linux_amd64/kn-linux-amd64.tar.gz && \
+    mv kn-macos-amd64.tar.gz /usr/share/kn/macos/kn-macos-amd64.tar.gz && \
+    mv kn-windows-amd64.zip /usr/share/kn/windows/kn-windows-amd64.zip
```
Comment (navidshaikh) [inline openshift/ci-operator/knative-images/client/Dockerfile.cliartifacts]: The input to `package_cliartifacts.sh` (i.e. cross platform binaries, LICENSE and the script itself) are fed from CI.  
See https://github.com/openshift/release/pull/7404/files#diff-072f710a93e689d38ab54a2831351f1eR22

**Comment by navidshaikh** [comment]
Steps to test this PR:
1 - Checkout this branch
2 - `make build-cross`
3 -  docker build
```
➜  knative-client git:(pr/cli-artifacts) ✗ docker build -t cli  -f openshift/ci-operator/knative-images/client/Dockerfile.cliartifacts  .
STEP 1: FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
STEP 2: RUN microdnf install -y zip tar gzip

(process:1): librhsm-WARNING **: 09:13:35.361: Found 0 entitlement certificates

(process:1): librhsm-WARNING **: 09:13:35.363: Found 0 entitlement certificates

(process:1): libdnf-WARNING **: 09:13:35.364: Loading "/etc/dnf/dnf.conf": IniParser: Can't open file
Downloading metadata...
Downloading metadata...
Downloading metadata...
Package                                                 Repository          Size
Installing:                                                                     
 gzip-1.9-9.el8.x86_64                                  ubi-8-baseos    170.9 kB
 tar-2:1.30-4.el8.x86_64                                ubi-8-baseos    858.4 kB
 unzip-6.0-41.el8.x86_64                                ubi-8-baseos    197.3 kB
 zip-3.0-23.el8.x86_64                                  ubi-8-baseos    276.8 kB
Transaction Summary:
 Installing:        4 packages
 Reinstalling:      0 packages
 Upgrading:         0 packages
 Removing:          0 packages
 Downgrading:       0 packages
Downloading packages...
Running transaction test...
Installing: unzip;6.0-41.el8;x86_64;ubi-8-baseos
Installing: zip;3.0-23.el8;x86_64;ubi-8-baseos
Installing: tar;2:1.30-4.el8;x86_64;ubi-8-baseos
Installing: gzip;1.9-9.el8;x86_64;ubi-8-baseos
Complete.
ae219d4e094c0f08ec4c5ae044686ba4db6db0b557537b2e2bfc0d0fd046c5df
STEP 3: ADD package_cliartifacts.sh .
d54a2248d72caeda7dec13851d88a71e6281fddcbe4bc6726a5ff512b8befeb9
STEP 4: ADD LICENSE .
96a1c3d1e88a9504a0a4eec50cadbbdad1f77f6cd69ea2fee2554397e902bcf6
STEP 5: ADD kn-*-amd64* .
4d8a54af61e9c99c7cf8c8dcd21ae7b481e4215e71d5f02a0891f14d0ef23f8e
STEP 6: RUN bash package_cliartifacts.sh
781e62ccd9e70c6c5b55b47da87c5c8d6b9d758c232bea75aa7316abe111a3f7
STEP 7: RUN mkdir -p /usr/share/kn/linux_amd64 &&     mkdir -p /usr/share/kn/macos &&     mkdir -p /usr/share/kn/windows &&     mv kn-linux-amd64.tar.gz /usr/share/kn/linux_amd64/ &&     mv kn-macos-amd64.tar.gz /usr/share/kn/macos/ &&     mv kn-windows-amd64.zip /usr/share/kn/windows/
7fc5963ee6ce4d8e9b571921b5b6806f954db78be04316fb80392e227f9bfd2e
STEP 8: LABEL       com.redhat.component="openshift-serverless-1-tech-preview-client-kn-rhel8-container"       name="openshift-serverless-1-tech-preview/client-kn-rhel8"       version="v0.12.0"       summary="Red Hat OpenShift Serverless 1 kn CLI"       description="CLI client 'kn' for managing Red Hat OpenShift Serverless 1"       io.k8s.display-name="Red Hat OpenShift Serverless 1 kn CLI"       io.k8s.description="CLI client 'kn' for managing Red Hat OpenShift Serverless 1"       io.openshift.build.source-location="https://github.com/openshift/knative-client"       maintainer="serverless-support@redhat.com"
STEP 9: COMMIT cli
99221f861a11764cfb9b1c7a32b7316940bcc1b0d1db91d307188068b51f56c7
99221f861a11764cfb9b1c7a32b7316940bcc1b0d1db91d307188068b51f56c7
```
4 -  Verify
```
➜  knative-client git:(pr/cli-artifacts) ✗ docker run  -t cli ls -l /usr/share/kn
total 12
drwxr-xr-x. 2 root root 4096 Mar  3 09:14 linux_amd64
drwxr-xr-x. 2 root root 4096 Mar  3 09:14 macos
drwxr-xr-x. 2 root root 4096 Mar  3 09:14 windows
➜  knative-client git:(pr/cli-artifacts) ✗ docker run  -t cli ls -l /usr/share/kn/linux_amd64
total 21340
-rw-r--r--. 1 root root 21851172 Mar  3 09:14 kn-linux-amd64.tar.gz
➜  knative-client git:(pr/cli-artifacts) ✗ docker run  -t cli ls -l /usr/share/kn/macos      
total 21128
-rw-r--r--. 1 root root 21632063 Mar  3 09:14 kn-macos-amd64.tar.gz
➜  knative-client git:(pr/cli-artifacts) ✗ docker run  -t cli ls -l /usr/share/kn/windows
total 21156
-rw-r--r--. 1 root root 21660822 Mar  3 09:14 kn-windows-amd64.zip
```

**Comment by navidshaikh** [comment]
/hold

these changes should be merged to master branch, not release-next.

**Comment by rhuss** [comment]
@navidshaikh I will take care

**Pull request closed without merging**